prody2/viewers/viewer_gnm.py

```diff
@@ -0,0 +1,266 @@
+                      label='Display Cross Correlation matrix?',
+                      help='Cross Correlation matrices are shown as heatmaps.')
+        
+
+    def _getVisualizeDict(self):
+        return {'displayModes': self._viewParam,
+                'displayMaxDistanceProfile': self._viewParam,
+                'displaySqFlucts': self._viewSQF,
+                'displayVmd': self._viewSingleMode,
+                'displayVmd2': self._viewSingleMode,
+                'displayDistanceProfile': self._viewSingleMode,
+                'displaySingleSqFluct': self._viewSQF,
+                'displayCovMatrix': self._viewMatrix,
+                'displayCrossCorrMatrix': self._viewMatrix
+                } 
+
+    def _viewMatrix(self, paramName):
+        """ visualisation for 2D covariance and Cross correlation matrices""" 
+        if paramName=='displayCovMatrix':
+            matrix = prody.parseArray(self.protocol.matrixFileCV.getFileName())
+            title = 'Covariance matrix'
+        else:
+            matrix = prody.parseArray(self.protocol.matrixFileCC.getFileName())
+            title = 'Cross Correlation matrix'
+
+        plotter = EmPlotter(mainTitle=title)
+        plot = prody.showMatrix(matrix, vmax=1, vmin=-1, origin='lower')
+        
+        return [plotter] 
+
+    def _viewParam(self, paramName):
+        if paramName == 'displayModes':
+            # The following two lines display modes.sqlite file
+            # modes =  self.protocol.outputModes
+            # return [ObjectView(self._project, modes.strId(), modes.getFileName())]
+            # The following two lines display modes.xmd file
+            if isinstance(self.protocol, SetOfNormalModes):
+                modes = os.path.dirname(self.protocol[1].getModeFile()) + ".xmd"
+            else:
+                modes =  self.protocol._getPath("modes.xmd")
+            return [DataView(modes)]
+
+        elif paramName == 'displayMaxDistanceProfile':
+            if isinstance(self.protocol, SetOfNormalModes):
+                fn = os.path.dirname(os.path.dirname(self.protocol[1].getModeFile())) + "/extra/maxAtomShifts.xmd"
+            else:
+                fn = self.protocol._getExtraPath("maxAtomShifts.xmd")
+            return [createShiftPlot(fn, "Maximum atom shifts", "maximum shift")]
+
+    def _viewSQF(self, paramName):
+        """ visualization of square fluctuation for all/single mode/s. """   
+        modes_path = self.protocol._getPath("modes.gnm.npz")
+        modes = prody.loadModel(modes_path)
+        
+        cumulSqFluct = self.cumulSqFluct.get()
+        modeNumber = self.modeNumber.get()
+
+        if paramName == 'displaySqFlucts':
+            plotter = EmPlotter()
+            plot = prody.showSqFlucts(modes)
+        
+        elif paramName == 'displaySingleSqFluct':
+            if cumulSqFluct:
+                plotter = EmPlotter(mainTitle="Qumulative")
+                plot = prody.showSqFlucts(modes[:modeNumber])
```

Comment:
This could cause a problem. You need to exclude the zero eigen value mode. That probably means checking if it has been calculated too.

Reply:
Again, it might be better to leave the option of range of modes to the user